### PR TITLE
[CDAP-13180] Initial PR for Namespace Details page

### PR DESCRIPTION
**Pull request by tonybach**
JIRA: https://issues.cask.co/browse/CDAP-13180

This PR sets up the Namespace Details page and implements most of the sections as in the mock. 

The 'Compute Profile' section is not implemented yet. This will come in the next PR.

### CODE REVIEW

File: cdap-ui/app/cdap/components/NamespaceDetails/NamespaceDetails.scss [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+
+@import "../../styles/variables.scss";
+
+.namespace-details-container {
+  /* 50 px top panel, 54px footer */
+  height: calc(100vh - 50px - 104px);
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/NamespaceDetails/NamespaceDetails.scss]: Is this necessary? we already have `#app-container` with height `calc(100vh - 104px)` right? Until `Home` I see all components taking the height of the parent.

File: cdap-ui/app/cdap/components/NamespaceDetails/NamespaceDetails.scss [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+              overflow: hidden;
+              text-overflow: ellipsis;
+              white-space: nowrap;
+            }
+          }
+        }
+      }
+    }
+
+    .namespace-details-mapping {
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/NamespaceDetails/NamespaceDetails.scss]: Shouldn't this be in `Mapping.scss` under the `Mapping` component?

File: cdap-ui/app/cdap/components/NamespaceDetails/NamespaceDetails.scss [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      margin-top: 10px;
+    }
+
+    .namespace-details-entity-count {
+      > span {
+        padding-right: 150px;
+      }
+    }
+
+    .namespace-details-preferences {
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/NamespaceDetails/NamespaceDetails.scss]: Similarly this in `Preferences.scss` under `Preferences` component?

**Review by ajainarayanan — COMMENTED** [review]
Just some minor comments. 👍 LGTM

File: cdap-ui/app/cdap/components/NamespaceDetails/store/ActionCreator.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      MyPreferenceApi.getNamespacePreferences({namespace}),
+      MyPreferenceApi.getNamespacePreferencesResolved({namespace}),
+      MyAppApi.list({namespace}),
+      MySearchApi.search(searchParams)
+    )
+    .subscribe(
+      (res) => {
+        let {name, description, config} = res[0];
+        let namespacePrefs = {...res[1]};
+        let resolvedPrefs = {...res[2]};
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/NamespaceDetails/store/ActionCreator.js]: You don't have to change it but you could do something like this,
```
let [ namespaceInfo, namespacePrefs, resolvedPrefs, pipelines, datasets ] = res;
```

Its a simple spread operator.

File: cdap-ui/app/cdap/components/NamespaceDetails/store/ActionCreator.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        let systemPrefs = {};
+        if (Object.keys(resolvedPrefs).length > Object.keys(namespacePrefs).length) {
+          Object.keys(resolvedPrefs).forEach(resolvedPrefKey => {
+            if (!(resolvedPrefKey in namespacePrefs)) {
+              systemPrefs[resolvedPrefKey] = resolvedPrefs[resolvedPrefKey];
+            }
+          });
+        }
+
+        let pipelineCount = res[3].filter(app => {
```
Comment (ajainarayanan) [inline cdap-ui/app/cdap/components/NamespaceDetails/store/ActionCreator.js]: We can already specify this in the query parameter for fetching only pipelines. We do this in the pipelines list view. We could avoid fetching all apps and filtering them in the UI.

**Review by ajainarayanan — COMMENTED** [review]
one minor comment.

File: cdap-ui/app/cdap/components/NamespaceDetails/store/ActionCreator.js (hunk 5 above)
Comment (elfenheart) [reply cdap-ui/app/cdap/components/NamespaceDetails/store/ActionCreator.js]: there is the custom apps count as well, so it's better to fetch all apps than making 2 api calls

**Comment by ajainarayanan** [comment]
👍 Code LGTM

**Comment by leacd** [comment]
LGTM

**Pull request merged**